code/modules/reagents/machinery/electrolyzer.dm

```diff
@@ -0,0 +1,346 @@
+		if(!state)
+			on = FALSE
+			playsound(src.loc, 'sound/machines/buzz-two.ogg', 50, 1 -3)
+			visible_message("\icon[src]\The [src] buzzes indicating that error has occured.")
+		else if(state == -1)
+			on = FALSE
+			playsound(src.loc, 'sound/machines/ping.ogg', 50, 1 -3)
+			visible_message("\icon[src]\The [src] pings indicating that process is complete.")
+		update_icon()
+		SSnano.update_uis(src)
+
+/obj/machinery/electrolyzer/attackby(obj/item/I, mob/user, params)
+	if(default_deconstruction(I, user))
+		return
+
+	if(default_part_replacement(I, user))
+		return
+
+	if(istype(I, /obj/item/weapon/reagent_containers) && I.is_open_container() && (!beaker || !separationBeaker))
+		. = TRUE //no afterattack
+		var/obj/item/weapon/reagent_containers/B = I
+		if(!user.unEquip(B, src))
+			return
+		if(!beaker)
+			beaker = B
+		else if(!separationBeaker)
+			separationBeaker = B
+		to_chat(user, SPAN_NOTICE("You add [B] to [src]."))
+		SSnano.update_uis(src)
+		update_icon()
+		return
+	return ..()
+
+/obj/machinery/electrolyzer/on_deconstruction()
+	if(beaker)
+		beaker.forceMove(get_turf(src))
+		beaker = null
+	if(separationBeaker)
+		separationBeaker.forceMove(get_turf(src))
+		separationBeaker = null
+	..()
+	
+
+/obj/machinery/electrolyzer/attack_hand(mob/user)
+	if(..())
+		return TRUE
+
+	user.set_machine(src)
+	ui_interact(user)
+
+/obj/machinery/electrolyzer/ui_interact(mob/user, ui_key = "main", datum/nanoui/ui = null, force_open = NANOUI_FOCUS)
+	var/list/data = ui_data()
+
+	// update the ui if it exists, returns null if no ui is passed/found
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		// the ui does not exist, so we'll create a new() one
+        // for a list of parameters and their descriptions see the code docs in \code\modules\nano\nanoui.dm
+		ui = new(user, src, ui_key, "electrolyzer.tmpl", name, 550, 400)
+		// when the ui is first opened this is the data it will use
+		ui.set_initial_data(data)
+		// open the new ui window
+		ui.open()
+
+
+
+/obj/machinery/electrolyzer/ui_data()
+	var/data = list()
+	data["on"] = on
+
+	if(beaker)
+		data["beaker"] = beaker.reagents.ui_data()
+	if(separationBeaker)
+		data["separationBeaker"] = separationBeaker.reagents.ui_data()
+	data["has_power"] = (stat & NOPOWER) ? FALSE : TRUE
+	return data
+
+
+/obj/machinery/electrolyzer/Topic(href, href_list)
+	if(..())
+		return
+
+	if(href_list["turn_on"])
+		on = TRUE
+		update_icon()
+
+	if(href_list["turn_off"])
+		on = FALSE
+		update_icon()
+
+	if(href_list["eject"] && beaker)
+		on = FALSE
+		if(beaker)
+			beaker.forceMove(get_turf(src))
+			beaker = null
+			update_icon()
+
+	if(href_list["ejectSecondary"] && beaker)
+		on = FALSE
+		if(separationBeaker)
+			separationBeaker.forceMove(get_turf(src))
+			separationBeaker = null
+		update_icon()
+
+	return 1 // update UIs attached to this object
+
+
+
+/obj/item/device/makeshiftElectrolyser
```

Comment:
Same stuff that was said for the centrifuge

Reply:
`makeshiftElectrolyser` is still not properly formatted.